model-optimizer/extensions/middle/SplitConcatPairToInterpolate.py

```diff
@@ -53,7 +53,16 @@ def get_concat_after_split(split: Node) -> Optional[Node]:
 
     dest = split.out_port(0).get_destinations()[0].node
     # The transformation is applicable, only if next node is Concat.
-    return dest if dest.soft_get('type') == 'Concat' else None
+    if dest.soft_get('type') != 'Concat':
+        return
+
+    # The transformation is applicable, only if Split is a unique producer for Concat.
+    dest_inputs = [p.get_connection().get_source().node for _, p in dest.in_ports().items() if not p.disconnected()]
+    names_of_concat_inputs = set([n.name for n in dest_inputs])
+    if len(names_of_concat_inputs) != 1:
+        return
+
+    return dest
```

Comment:
do we need to return `dest`? Previously it was not needed. And also it is not aligned with other `returns`

Reply:
Yes, we need, and previously it was needed: see https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/extensions/middle/SplitConcatPairToInterpolate.py#L56. Next, in other `return`s we return `None`, and we check the returned result (is it result `None` or is it not `None`) in the function `get_interpolate_pattern` (see https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/extensions/middle/SplitConcatPairToInterpolate.py#L64).

Reply:
Finally, there are type annotation `def get_concat_after_split(split: Node) -> Optional[Node]`. And, hence, this function can return either `Node`, or `None`.